src/utils/network.py

```diff
@@ -76,19 +64,75 @@ def is_default_route(r):
 
     if not any(is_default_route(r) for r in routes):
         logger.debug(
-            "No default route for IPv%s, so assuming there's no IP address for that version"
-            % protocol
+            f"No default route for IPv{protocol}, so assuming there's no IP address for that version"
         )
         return None
 
-    url = "https://ip%s.yunohost.org" % (protocol if protocol != 4 else "")
-    logger.debug("Fetching IP from %s " % url)
+    # Check URLS
+    ip_list = get_public_ips(protocol)
+    if len(ip_list)>0:
+        return ip_list[0]
 
-    try:
-        return download_text(url, timeout=30).strip()
-    except Exception as e:
-        logger.debug(f"Could not get public IPv{protocol} : {e}")
-        return None
+    return None
+
+def get_public_ips(protocol=4):
```

Comment:
This is getting confusing because we end up with `get_public_ip` calling `get_public_ip_from_remote_server` calling `get_public_ips` ... I would tend to keep `get_public_ip_from_remote_server` and `get_public_ips` merged together as one big function (yeah) because it's easier to read and understand. Maybe renaming the function into `get_public_ip_from_mirrors` in the process